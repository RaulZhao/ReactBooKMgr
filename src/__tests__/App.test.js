import React from 'react';
import ReactDOM from 'react-dom';
import App, { Search, BookList } from '../App';
import renderer from 'react-test-renderer';
import mockData from '../dummy_book';

describe('App Test', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<App {...mockData}/>, div);
  });

  test('App snapshots', () => {
    const component = renderer.create(
      <App {...mockData}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
