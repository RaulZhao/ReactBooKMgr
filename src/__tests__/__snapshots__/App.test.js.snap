// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test App snapshots 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <div
    className="book-list-container"
  >
    <h3>
      Books List:
    </h3>
  </div>
  <div
    className="search-input"
  >
    <span>
      Search Book:
    </span>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="search-result"
  >
    <h4 />
  </div>
</div>
`;
